Azure/AgentBaker
# bug: compress customdata

**What type of PR is this?**

/kind bug


**What this PR does / why we need it**:

This PR fixes a bug where the custom data we put on a node to bootstrap the node can be too large. It consists of two changes:

* gzip the entire custom data before base-64 encoding the file
* increase gzip compression level from default to "best compression"

From one of the e2e tests:

Size of custom data:
* Before gzip: 83252 bytes
* Increasing compression level: 83092 bytes
* With gzip and better compression: 57248 bytes

Oddly, some of the files inside the custom data are already gzipped. When we removed that inner compression, the overall file size did not drop much (81852 bytes). So we're going with the base64 encoded gzip of a file that contains base64 encoded gzip files. It surprised us too.

To test this worked, we:
* Ran e2e tests to check that nodes could be created in the agent baker e2e test environment.
* Created a standalone environment, versioned this branch into the standlone, created a cluster, and checked the cloud-init directory to verify that a gzipped file was put there.

![image](https://github.com/user-attachments/assets/7988e4da-4505-4d4b-82e5-93b0315337c4)
![image](https://github.com/user-attachments/assets/4d960c54-7af7-4f57-adfb-2e28ff17ffd7)


**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
Improve compression of custom data that's sent to the VM for bootstrapping.
```
